tools/opt/opt.cpp

```diff
@@ -225,14 +287,24 @@ static void AddOptimizationPasses(PassManagerBase &MPM,FunctionPassManager &FPM,
   Builder.SLPVectorize =
       DisableSLPVectorization ? false : OptLevel > 1 && SizeLevel < 2;
 
+  // Add target-specific passes that need to run as early as possible.
+  if (TM)
+    Builder.addExtension(
+        PassManagerBuilder::EP_EarlyAsPossible,
+        [&](const PassManagerBuilder &, legacy::PassManagerBase &PM) {
+          TM->addEarlyAsPossiblePasses(PM);
+        });
+
+  if (Coroutines)
+    addCoroutinePassesToExtensionPoints(Builder);
+
   Builder.populateFunctionPassManager(FPM);
   Builder.populateModulePassManager(MPM);
 }
 
-static void AddStandardLinkPasses(PassManagerBase &PM) {
-  llvm_seahorn::PassManagerBuilder Builder;
+static void AddStandardLinkPasses(legacy::PassManagerBase &PM) {
+  PassManagerBuilder Builder;
```

Comment:
It should  be `llvm_seahorn::PassManagerBuilder`. Otherwise, we will use the llvm passes.